juriscraper/opinions/united_states/state/nh.py

```diff
@@ -1,127 +1,150 @@
 
-class Site(OpinionSite):
+
+class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.url = "http://www.courts.state.nh.us/supreme/opinions/{current_year}/index.htm".format(
-            current_year=date.today().year
-        )
         self.court_id = self.__module__
-        self.link_path = 'id("content")/div//ul//li//a[position()=1]'
-        self.link_text_regex = re.compile(
-            r"(\d{4}-\d+(?!.*\d{4}-\d+))(?:,|\.|\s?)\s+(.*)", re.M
-        )
-        self.docket_name_pairs = False
+        # Parameters are included in the URL because this court's API only
+        # allows GET requests. Current architecture for scrapers only includes
+        # parameters in requests with the POST method
+        self.url = "https://www.courts.nh.gov/content/api/documents?sort=field_date_posted%7Cdesc&page=1&size=30&purpose=1331"
+        self.status = "Published"
 
-    def _get_case_names(self):
-        self.docket_name_pairs = self._get_anchor_docket_name_pairs()
-        return [pair["name"] for pair in self.docket_name_pairs]
-
-    def _get_download_urls(self):
-        path = "id('content')/div//ul//li//a[position()=1]"
-        download_url = []
-        for element in self.html.xpath(path):
-            url = element.xpath("./@href")[0]
-            download_url.extend([url])
-        return download_url
-
-    def _get_case_dates(self):
-        dates = []
-        path = 'id("content")/div//strong'
-        sub_path = (
-            "./following-sibling::ul[1]//li|../following-sibling::ul[1]//li"
-        )
-        for element in self.html.xpath(path):
-            text = element.text_content().strip().rstrip(":")
-            try:
-                case_date = convert_date_string(text)
-            except ValueError:
-                # Sometimes the court includes "To be released..."
-                # sections, without links, which we skip here
+    def _process_html(self) -> None:
+        for case in self.html["data"]:
+            content = fromstring(case["documentContent"])
+            urls = content.xpath(
+                ".//div[@class='document__detail__title']//a/@href"
+            )
+            if not urls:
+                logger.info(f"Opinion without URL to file: {case}")
                 continue
-            dates.extend([case_date] * len(element.xpath(sub_path)))
-        return dates
-
-    def _get_precedential_statuses(self):
-        return ["Published"] * len(self.case_names)
 
-    def _get_docket_numbers(self):
-        return [pair["docket"] for pair in self.docket_name_pairs]
+            date = case["documentPosted"]
+            if not date:
+                logger.info(f"Opinion without date: {case}")
+                continue
 
-    def _get_anchor_docket_name_pairs(self):
-        """The court has some ugly HTML practices that we need to handle.
+            name, docket = self.name_docket(case["documentName"])
```

Comment:
You should just replace this line and method with the following.

```suggestion
            # In juvenile cases - the case name contains the docket number
            # So we use a negative lookbehind to ignore docket numbers after
            # the word juvenile.
            dockets = re.findall(
                r"(?<!JUVENILE )((\w+-)?\d{4}-\d+)", case["documentName"]
            )
            docket = ", ".join([docket[0] for docket in dockets])

            # Strip out the docket numbers and clean up the case names.
            name = re.sub(
                r"(?<!JUVENILE )((\w+-)?\d{4}-\d+)", "", case["documentName"]
            )
            name = name.strip(", ").replace(", and ,", " and ")

```

Reply:
Used code in new version for method `self.name_docket` to handle some edge cases.